src/overlays/actors/ovl_Bg_Ice_Shutter/z_bg_ice_shutter.c

```diff
@@ -27,19 +31,105 @@ const ActorInit Bg_Ice_Shutter_InitVars = {
     (ActorFunc)BgIceShutter_Update,
     (ActorFunc)BgIceShutter_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Ice_Shutter/func_80891AC0.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Ice_Shutter/BgIceShutter_Init.s")
+extern UNK_TYPE D_06002854;
+extern Gfx D_06002740[];
+
+static InitChainEntry sInitChain[] = {
+    ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_STOP),
+};
+
+void func_80891AC0(BgIceShutter* thisx) {
+    f32 sp24;
+
+    sp24 = Math_Sins(thisx->dyna.actor.shape.rot.x) * thisx->dyna.actor.velocity.y;
+    thisx->dyna.actor.posRot.pos.y =
+        (Math_Coss(thisx->dyna.actor.shape.rot.x) * thisx->dyna.actor.velocity.y) + thisx->dyna.actor.initPosRot.pos.y;
+    thisx->dyna.actor.posRot.pos.x =
+        (Math_Sins(thisx->dyna.actor.shape.rot.y) * sp24) + thisx->dyna.actor.initPosRot.pos.x;
+    thisx->dyna.actor.posRot.pos.z =
+        (Math_Coss(thisx->dyna.actor.shape.rot.y) * sp24) + thisx->dyna.actor.initPosRot.pos.z;
+}
+
+void BgIceShutter_Init(Actor* thisx, GlobalContext* globalCtx) {
+    BgIceShutter* this = THIS;
+    f32 sp24;
+    s32 localC;
+    s32 sp28;
+    f32 temp_f6;
+
+    localC = 0;
+    Actor_ProcessInitChain(thisx, sInitChain);
```

Comment:
so can you use this->dyna.actor instead of thisx? it's prefered to access everything through `this` if possible